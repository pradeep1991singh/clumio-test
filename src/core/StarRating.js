import React, { useState } from "react";

const StarRating = ({ onRatingChange }) => {
  const [rating, setRating] = useState(0);
  const [hover, setHover] = useState(0);

  const handleRateChange = (rating) => {
    setHover(rating);
    onRatingChange(rating);
  };

  return (
    <div className="star-rating">
      {[...Array(5)].map((star, index) => {
        index += 1;
        return (
          <button
            type="button"
            key={index}
            className={
              index <= (hover || rating)
                ? "text-orange-500"
                : "dark:text-gray-500"
            }
            onClick={() => setRating(index)}
            onMouseEnter={() => setHover(index)}
            onMouseLeave={() => handleRateChange(rating)}
          >
            <span className="bg-transparent border-none outline-one cursor-pointer">
              &#9733;
            </span>
          </button>
        );
      })}
    </div>
  );
};

export default StarRating;
