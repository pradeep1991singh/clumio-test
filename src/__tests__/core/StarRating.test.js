import React from "react";
import { render } from "@testing-library/react";
import StarRating from '../../core/StarRating';

const defaultProps = {
  onChange: jest.fn(),
};

describe("<StarRating />", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<StarRating {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});