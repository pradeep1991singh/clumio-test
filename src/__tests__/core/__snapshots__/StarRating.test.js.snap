// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="star-rating"
  >
    <button
      class="dark:text-gray-500"
      type="button"
    >
      <span
        class="bg-transparent border-none outline-one cursor-pointer"
      >
        ★
      </span>
    </button>
    <button
      class="dark:text-gray-500"
      type="button"
    >
      <span
        class="bg-transparent border-none outline-one cursor-pointer"
      >
        ★
      </span>
    </button>
    <button
      class="dark:text-gray-500"
      type="button"
    >
      <span
        class="bg-transparent border-none outline-one cursor-pointer"
      >
        ★
      </span>
    </button>
    <button
      class="dark:text-gray-500"
      type="button"
    >
      <span
        class="bg-transparent border-none outline-one cursor-pointer"
      >
        ★
      </span>
    </button>
    <button
      class="dark:text-gray-500"
      type="button"
    >
      <span
        class="bg-transparent border-none outline-one cursor-pointer"
      >
        ★
      </span>
    </button>
  </div>
</DocumentFragment>
`;
